x/reporter/reporter.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reporter
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"math/rand"
+	"net/http"
+)
+
+var debugLog log.Logger = *log.New(io.Discard, "", 0)
+var httpClient = &http.Client{}
+
+func sendReport(record map[string]interface{}, collectorURL string) error {
```

Comment:
I changed the type to `any` but it might be safer to define a more specific type. I guess we would also need to accept slice/array of JSON objects to be reported as a batch, e.g. `[{}, {}]`. In that case, the reporter would have to iterate through the slice and send each JSON record one by one.